kommunicate/src/main/java/com/applozic/mobicomkit/api/conversation/database/MessageDatabaseService.java

```diff
@@ -1206,7 +1206,7 @@ public void deleteChannelConversation(Integer channelKey) {
         Utils.printLog(context, TAG, "Delete " + deletedRows + " messages.");
     }
 
-    public synchronized void updateContactUnreadCount(String userId) {
+    public void updateContactUnreadCount(String userId) {
         try {
```

Comment:
Removing synchronized might create issues, as it is reading values and incrementing. Two threads might be reading the same value and updating the count by 1.

Reply:
As per the [documentation](http://developer.android.com/reference/android/database/sqlite/SQLiteDatabase.html#setLockingEnabled(boolean)), access to the SQLiteDatabase is synchronized by default, thus guaranteeing that no two threads will ever touch it at the same time.

Reply:
We do not need to synchronize manually